frontend/src/metabase-lib/Question.ts

```diff
@@ -1286,6 +1298,211 @@ class QuestionInner {
+          components.push(DESCRIPTION.formatField(fieldDef, options), " → ");
+        }
+      }
+      components.push(DESCRIPTION.formatField(target.field, options));
+      if (target.unit) {
+        components.push(` (${target.unit})`);
+      }
+      return components;
+    } catch (e) {
+      console.warn(
+        "Couldn't format field name for field",
+        field,
+        "in table",
+        tableMetadata,
+      );
+    }
+    return "[Unknown Field]";
+  }
+
+  private _getTableDescription(tableMetadata) {
+    return [inflection.pluralize(tableMetadata.display_name)];
+  }
+
+  private _getAggregationDescription(tableMetadata, query, options) {
+    return DESCRIPTION.conjunctList(
+      QUERY.getAggregations(query).map(aggregation => {
+        if (AGGREGATION.hasOptions(aggregation)) {
+          if (AGGREGATION.isNamed(aggregation)) {
+            return [AGGREGATION.getName(aggregation)];
+          }
+          aggregation = AGGREGATION.getContent(aggregation);
+        }
+        if (AGGREGATION.isMetric(aggregation)) {
+          const metric = _.findWhere(tableMetadata.metrics, {
+            id: AGGREGATION.getMetric(aggregation),
+          });
+          return metric ? metric.name : "[Unknown Metric]";
+        }
+        switch (aggregation[0]) {
+          case "rows":
+            return [t`Raw data`];
+          case "count":
+            return [t`Count`];
+          case "cum-count":
+            return [t`Cumulative count`];
+          case "avg":
+            return [
+              t`Average of `,
+              this._getFieldName(tableMetadata, aggregation[1], options),
+            ];
+          case "median":
+            return [
+              t`Median of `,
+              this._getFieldName(tableMetadata, aggregation[1], options),
+            ];
+          case "distinct":
+            return [
+              t`Distinct values of `,
+              this._getFieldName(tableMetadata, aggregation[1], options),
+            ];
+          case "stddev":
+            return [
+              t`Standard deviation of `,
+              this._getFieldName(tableMetadata, aggregation[1], options),
+            ];
+          case "sum":
+            return [
+              t`Sum of `,
+              this._getFieldName(tableMetadata, aggregation[1], options),
+            ];
+          case "cum-sum":
+            return [
+              t`Cumulative sum of `,
+              this._getFieldName(tableMetadata, aggregation[1], options),
+            ];
+          case "max":
+            return [
+              t`Maximum of `,
+              this._getFieldName(tableMetadata, aggregation[1], options),
+            ];
+          case "min":
+            return [
+              t`Minimum of `,
+              this._getFieldName(tableMetadata, aggregation[1], options),
+            ];
+          default:
+            return [formatExpression(aggregation, { tableMetadata })];
+        }
+      }),
+      "and",
```

Comment:
Should we translate this?

Reply:
I've added TODOs to the various untranslated strings. I'll try to find a more i18n-friendly way to do these text lists. Not all languages structure lists the same way, of course. Even the above translated strings `"Maximum of "` etc. are naively concatenated rather than using i18n placeholders, so they assume a certain syntax. We can do better!